[main] distro/rhel9: edge images default to LVM

7flying opened the pull request:
Change partition tables on `edgeBase` images to use 'LVM partitioning'.
We need to ensure that LVM stages are done before LUKS stages (e.g. remove-key) or the pipelines will break (we cannot open a device when its password has changed).

This PR requires changes on `coreos-installer-dracut/dracut/scripts/coreos-installer-growfs`([PR](https://github.com/coreos/coreos-installer-dracut/pull/9)) required by rhel90-edge images to account for the change from traditional partitioning to LVM, otherwise the image will boot on an emergency shell and will need a manual reboot. ~~This is why this is marked as a draft PR.~~

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

gicmo commented:
LGTM on a first glance!

henrywang commented:
@7flying Thanks for doing this! :+1: 
Is this change only on RHEL 9.1? No LVM on RHEL 8.7? Thanks.

7flying commented:
@henrywang I will send changes 8.7 on a different PR!

7flying commented:
Since @gicmo's last comment I've added a test and modified the partition table to set 9GB to the rootlv on the edge images. I've also linked the relevant PR to coreos-installer-dracut.

henrywang commented:
@7flying This PR needs a new `coreos-installer` release which include MR https://gitlab.com/redhat/centos-stream/rpms/rust-coreos-installer/-/merge_requests/16? Thanks.

runcom commented:
@achilleas-k can you review this and https://github.com/osbuild/osbuild-composer/pull/2904 - these are required for us to ship lvm images

achilleas-k reviewed (changes requested):
Looks good, but we're going to need manifest updates.  You can run 
```
./tools/test-case-generators/generate-all-test-cases \
    --output test/data/manifests \
    --image-type edge-raw-image \
    manifests
```
to generate new ones.

runcom commented:
@achilleas-k I'm not sure the test failure is related here (but have no familiarity with the failure :))

runcom commented:
This looks good now @achilleas-k (only f35aarch64 is still building) ptal

achilleas-k reviewed (approved):
All good now.  Thank you.

runcom commented:
> @runcom LVM testing updated.

they're still failing now

henrywang commented:
@runcom All green except simplified installer.

runcom commented:
> @runcom All green except simplified installer.

we just a new working snapshot

runcom commented:
I'll remove cs9 tests tomorrow to make this happy

henrywang commented:
@runcom @achilleas-k The only failure is `aws.sh` test on aarch64 which is not caused by this PR. Can we land this PR? Thanks.

ondrejbudai commented:
@runcom Can you rebase, please? I have no idea what's wrong, GitHub doesn't allow me to merge without rebasing. :shrug:

henrywang commented:
> @runcom Can you rebase, please? I have no idea what's wrong, GitHub doesn't allow me to merge without rebasing. shrug

Thanks @ondrejbudai. Rebased.

ondrejbudai reviewed (approved):
Let's get this in!
